### ISSUE TITLE
RuboCop::Cop::Documentation.url_for returns nil for builtin Cops with no instance methods defined

### ISSUE THREAD

**Issue by Fryguy** [rubocop/rubocop#15048]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Other

### Cop name

Layout/AssignmentIndentation, Lint/UnusedBlockArgument, Metrics/CyclomaticComplexity, Metrics/AbcSize, Metrics/PerceivedComplexity, Style/HashExcept, Style/HashSlice

### Steps to reproduce the problem

These cops have no instance methods defined (cop.instance_methods(false).empty? == true), which causes RuboCop::Cop::Documentation.builtin?(cop) to return false on line 59 of lib/rubocop/cop/documentation.rb. As a result, url_for returns nil on line 21.

```ruby
RuboCop::Cop::Documentation.url_for(RuboCop::Cop::Layout::AssignmentIndentation)
# => nil
```

### Expected behavior

```ruby
RuboCop::Cop::Documentation.url_for(RuboCop::Cop::Layout::AssignmentIndentation)
# => "https://docs.rubocop.org/rubocop/cops_layout.html#layoutassignmentindentation"
```

### Actual behavior

```ruby
RuboCop::Cop::Documentation.url_for(RuboCop::Cop::Layout::AssignmentIndentation)
# => nil
```

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
$ bundle exec rubocop -V
1.86.0 (using Parser 3.3.9.0, rubocop-ast 1.49.1, analyzing as Ruby 2.7, running on ruby 3.3.10) [arm64-darwin24]
  - rubocop-performance 1.26.1
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.9.0
```
